atc/metric/emitter/lager.go

```diff
@@ -5,20 +5,24 @@ import (
 
 	"code.cloudfoundry.org/lager"
 	"github.com/concourse/concourse/atc/metric"
+	"github.com/pkg/errors"
 )
 
 type LagerEmitter struct{}
 
 type LagerConfig struct {
-	Enabled bool `long:"emit-to-logs" description:"Emit metrics to logs."`
-}
-
-func init() {
-	metric.Metrics.RegisterEmitter(&LagerConfig{})
+	Enabled bool `yaml:"enabled,omitempty" env:"CONCOURSE_LAGER_ENABLE,CONCOURSE_EMIT_TO_LOGS"`
 }
 
+func (config *LagerConfig) ID() string          { return "lager" }
 func (config *LagerConfig) Description() string { return "Lager" }
-func (config *LagerConfig) IsConfigured() bool  { return config.Enabled }
+func (config *LagerConfig) Validate() error {
+	if !config.Enabled {
+		return errors.New("enabled needs to be true")
```

Comment:
does this even need to be here? seems like it could just always be valid, and we only use it if it's enabled?

Reply:
Oh so have it not fail during validation phase but push the checking of `enabled` to the initializing of the emitter?